### PR TITLE
feat: [sc-26755] Add support for content field in transformer

### PR DESCRIPTION
**Pull request by ignatospadov**
Story details: https://app.shortcut.com/cheddar/story/26755

### CODE REVIEW

**Comment by ignatospadov** [comment]
Can you take a look when you have a chance over the embed types and see if there are any potential conflicts that might affect the apps? We will have to modify some styling and css to support the iframes but it should be minimal. I tried to maintain the same structure for styling except when it made more sense to allow an iframe (this will need to have styling done in various apps i imagine but it should be minimal) @zhiln

File: src/parser/contentful/parser.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { assetResolver, embedResolver, listItemResolver, listResolver, paragraphResolver, quoteResolver, tableCellResolver, tableHeaderCellResolver } from '../../resolvers/contentful'
```
Comment (zhiln) [inline src/parser/contentful/parser.js:1]: run prettier?

File: src/resolvers/contentful/common.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export const removeParagraphNode = (node) => {
```
Comment (zhiln) [inline src/resolvers/contentful/common.js:1]: is this about that weird empty paragraph node?

File: src/resolvers/contentful/image.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export default (node) => `<div><img src=\"https:${node.data.target.fields.file.url}\" alt=\"\"/><em>${node.data.target.fields.description}</em></div>`
```
Comment (zhiln) [inline src/resolvers/contentful/image.js:1]: Maybe break this into multiple lines for readability?

File: src/resolvers/contentful/paragraph.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import config from '../../config'
+import shortcodeResolver from './shortcode.js'
+
+const isShortCode = (node) => {
+  if (node.content[0].nodeType === 'text' && node.content[0].value.trim().startsWith('[') && node.content[0].value.trim().endsWith(']')) {
```
Comment (zhiln) [inline src/resolvers/contentful/paragraph.js:5]: This will recognize hyperlink as shortCode as well?

File: src/resolvers/contentful/shortcode.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      return spotimResolver(shortcode)
+    case 'twitter':
+      return twitterResolver(shortcode)
+    case 'youtube':
+      return youtubeResolver(shortcode)
+    case 'ugc-form':
+      return formResolver(shortcode)
+    case 'ugc-gallery':
+      return galleryResolver(shortcode)
+    case null:
```
Comment (zhiln) [inline src/resolvers/contentful/shortcode.js:20]: is it necessary to handle `undefined` and `null`? it should all fell into default case. Unless you want to handle them in the future

File: src/util/shortcode-parser.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
+export default function (shortCode) {
```
Comment (zhiln) [inline src/util/shortcode-parser.js:1]: I would provide some short description on what is going on here and why code style is different, maybe the source too

File: src/resolvers/contentful/youtube.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import config from '../../config'
+
+function extract(url){
```
Comment (zhiln) [inline src/resolvers/contentful/youtube.js:3]: vars and semicolons? It's evil 😁

**Review by zhiln — COMMENTED** [review]
Good work, @ignatospadov!

I looked at it and couldn't find any potential conflicts. I think if there will be any once we work on frontends, we can always fix them on the go. 

I know you're still working on it, but I left few comments also.

File: src/parser/contentful/parser.js (hunk 1 above)
Comment (ignatospadov) [reply src/parser/contentful/parser.js:1]: will do thanks

File: src/resolvers/contentful/common.js (hunk 2 above)
Comment (ignatospadov) [reply src/resolvers/contentful/common.js:1]: RTE likes to nest lists inside paragraphs along with a few other types. This is just for elimitating nesting within paragraphs when it doesent make sense/breaks out existing structure too much

File: src/resolvers/contentful/paragraph.js (hunk 4 above)
Comment (ignatospadov) [reply src/resolvers/contentful/paragraph.js:5]: will do

File: src/util/shortcode-parser.js (hunk 6 above)
Comment (ignatospadov) [reply src/util/shortcode-parser.js:1]: Yeah this is a copypasta from a gist. It works and i dont feel like rewriting the thing but ill add more info from here https://gist.github.com/jeffdrumgod/db908ac2e4623a3f586f60a119c9b772

File: src/resolvers/contentful/youtube.js (hunk 7 above)
Comment (ignatospadov) [reply src/resolvers/contentful/youtube.js:3]: copypasta XD

File: src/resolvers/contentful/paragraph.js (hunk 4 above)
Comment (ignatospadov) [reply src/resolvers/contentful/paragraph.js:5]: No. This only checks if text inside paragraph starts with the shortcode syntax and ends with the closing bracket

Comment (zhiln) [reply src/resolvers/contentful/paragraph.js:5]: oh, right. HyperLink has its own nodeType.

**Comment by ignatospadov** [comment]
Ready for review @zhiln @petro-bishko-sigma 🙏

**Review by petro-bishko-sigma — APPROVED** [review]
LGTM!

**Review by zhiln — APPROVED** [review]
👍

**Pull request merged**